tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/method/LabelSetCreatePerformer.scala

```diff
@@ -62,21 +65,16 @@ class LabelSetCreatePerformer @Inject()(labelRepository: LabelRepository,
     LabelCreationRequest.validateProperties(jsObject)
       .flatMap(validJsObject => Json.fromJson(validJsObject)(LabelSerializer.labelCreationRequest) match {
         case JsSuccess(creationRequest, _) => Right(creationRequest)
-        case JsError(errors) => Left(LabelCreationParseException(labelSetError(errors)))
+        case JsError(errors) => Left(LabelCreationParseException(standardError(errors)))
       })
 
-  private def labelSetError(errors: collection.Seq[(JsPath, collection.Seq[JsonValidationError])]): SetError =
-    errors.head match {
-      case (path, Seq()) => SetError.invalidArguments(SetErrorDescription(s"'$path' property in Label object is not valid"))
-      case (path, Seq(JsonValidationError(Seq("error.path.missing")))) => SetError.invalidArguments(SetErrorDescription(s"Missing '$path' property in Label object"))
-      case (path, Seq(JsonValidationError(Seq(message)))) => SetError.invalidArguments(SetErrorDescription(s"'$path' property in Label object is not valid: $message"))
-      case (path, _) => SetError.invalidArguments(SetErrorDescription(s"Unknown error on property '$path'"))
-    }
-
   private def createLabel(mailboxSession: MailboxSession,
                           clientId: LabelCreationId,
                           labelCreationRequest: LabelCreationRequest): SMono[LabelCreationResult] =
     SMono.fromPublisher(labelRepository.addLabel(mailboxSession.getUser, labelCreationRequest))
       .map(label => LabelCreationSuccess(clientId, LabelCreationResponse(label.id, label.keyword)))
+      .flatMap(creationSuccess => SMono(labelChangesRepository.save(LabelChange.created(
+        mailboxSession.getUser, stateFactory, creationSuccess.labelCreationResponse.id)))
+        .`then`(SMono.just[LabelCreationResult](creationSuccess)))
```

Comment:
IMO extract this to a method e.g `populateLabelChanges` would be more clear.

E.g: 
```java
createLabel(...)
  .populateLabelChanges(...)
```

Reply:
BTW maybe we can batch all the created + updated + destroyed changes into a single `LabelChange` object and then just 1 query to DB, instead of 3 as it stands now?

Reply:
good catch, it really better

Reply:
Sorry, I reviewed it quickly and missed this point.

I do not see this solved. (still 3 DB / `labelChangesPopulate.populate` requests).
Could created + updated + destroyed changes be combined and populated only 1?

Reply:
I think we can extract created, updated, destroyed results all at once at `LabelSetMethod`, no?

```scala
for {
      created <- createPerformer.createLabels(mailboxSession, request)
      updated <- updatePerformer.update(request, mailboxSession.getUser)
      destroyed <- deletePerformer.deleteLabels(request, mailboxSession)
} yield {
      extractAndCombinedChanges(created, updated, destroyed)
}
```